packages/api/src/routes/studies.ts

```diff
@@ -60,41 +68,61 @@ router.post("/new", isAuthenticated, async (req, res, next) => {
 
 router.put("/:id", isAuthenticated, async (req, res, next) => {
   try {
-    const study = await Study.findOneAndUpdate(
-      { _id: req.params["id"] },
-      req.body,
-      { new: true }
-    );
-    res.json(study);
+    const user = req.user as HydratedDocument<IUser>;
+    const studyId = req.params["id"];
+    const studyFromUser = user.studies.find((id) => id.toString() === studyId);
+    const studyFromDB = await Study.findById(studyId);
+
+    if (!studyFromUser) {
+      return next(new HttpError(404));
+    } else if (studyFromDB && studyFromDB.owner !== user) {
+      return next(new HttpError(403));
+    } else if (!studyFromDB) {
+      const newStudy = await Study.create({});
+      await user.updateOne({ $push: { studies: newStudy._id } });
+      res.status(201).json(newStudy);
+    } else {
+      const study = await Study.findOneAndUpdate(
+        { _id: req.params["id"] },
+        req.body,
+        { new: true }
+      );
+      res.json(study);
+    }
   } catch (e) {
     next();
   }
 });
 
-router.put("/:studyId/tasks/:taskId", async (req, res, next) => {
-  try {
-    const study = await Study.findOne({ _id: req.params["studyId"] });
-    if (!study) {
-      return next(new HttpError(404));
-    }
-
-    const task = await CustomizedBattery.findOneAndUpdate(
-      { _id: req.params["taskId"] },
-      req.body,
-      {
-        upsert: true,
-        new: true,
+router.put(
+  "/:studyId/tasks/:taskId",
```

Comment:
I think there's a case here that's not covered:
* user 1:
  * study 1:
    * task 1
* user 2:
  * study 2:
    * task 2

user 1 hits `PUT /studies/1/tasks/2`, seems like they are able to bypass the security and update task 2 (i think) (probably worth adding a test for this case)